File: urbanairship-core/src/main/java/com/urbanairship/iam/ImmediateDisplayCoordinator.java
package com.urbanairship.iam;

import androidx.annotation.NonNull;

/**
 * Display coordinator that is always ready.
 */
class ImmediateDisplayCoordinator extends DisplayCoordinator {

    @Override
    public boolean isReady() {
        return true;
    }

    @Override
    public void onDisplayStarted(@NonNull InAppMessage message) {}

    @Override
    public void onDisplayFinished(@NonNull InAppMessage message) {}

}

File: urbanairship-automation/src/main/java/com/urbanairship/iam/coordinator/ImmediateDisplayCoordinator.kt
/* Copyright Airship and Contributors */

package com.urbanairship.iam.coordinator

import com.urbanairship.app.ActivityMonitor
import com.urbanairship.iam.InAppMessage
import kotlinx.coroutines.flow.StateFlow

internal class ImmediateDisplayCoordinator(
    activityMonitor: ActivityMonitor,
    private val activityTracker: DisplayActivityTracker
) : DisplayCoordinator {

    override val isReady: StateFlow<Boolean> = activityMonitor.foregroundState

    override fun messageWillDisplay(message: InAppMessage) {
        activityTracker.messageWillDisplay()
    }

    override fun messageFinishedDisplaying(message: InAppMessage) {
        activityTracker.messageFinishedDisplaying()
    }
}
